### DISCUSSION TITLE
Exporting generated payloads and bytes?

### DISCUSSION THREAD

**Discussion by ar3910** [jtpereyda/boofuzz#725] [Q&A]
Hello everyone, 

I wanted to know if there's a way for boofuzz to export generated payloads/bytes to an external file (or function) instead of directly transmitting it over to the target?

Some extra detail:
I developed my own little tool for SOME/IP fuzzing and it works fine over IPv4. However, I quickly realized Boofuzz does support IPv6 so I cannot fuzz targets with an IPv6 using my tool. Now I'm thinking if I could just forward the generated packets/payloads by Boofuzz to an external file or function, then I can use some other tool or script to read from it and transmit the packets over IPv6 using some other tooling. This way we can also fuzz targets running IPv6 as well.

Any tips or suggestions are welcome. 

Thanks!

**Comment by SR4ven** [comment]
Hi @ar3910!

Take a look at https://github.com/jtpereyda/boofuzz/blob/master/boofuzz/fuzz_logger_csv.py. You could use it to save the fuzz data while targeting a dummy target.
The only problem with this could be that the test data gets truncated but I'm not sure if that also applies to the CSV logger.
Alternatively you could also write your own logger.

**Comment by ar3910** [comment]
Thanks for the reply.
We are doing something very similar to this already using the boofuzz-result database. 

I wanted to know if it is possible to achieve this using the "pre_send_callback" or "post_send_callback" also?

**Reply by SR4ven** [reply]
Yes that should be possible too. The `session` object is passed to the callback which should hold the fuzz data.
However, you should perhaps prefer a Monitor instead since the pre and post_send_callback options are kind of deprecated. See https://github.com/jtpereyda/boofuzz/blob/ec4dfb5f5ac2d170fdc9f9418e28544e563479a6/boofuzz/monitors/callback_monitor.py

Another option would be to write your own target implementing `ITargetConnection` which executes whatever code you need instead of actually sending the data over a socket.
